include/z64camera.h

```diff
@@ -0,0 +1,389 @@
+    /* 0x45 */ CAM_FUNC_SPECIAL8,
+    /* 0x46 */ CAM_FUNC_SPECIAL9,
+    /* 0x47 */ CAM_FUNC_MAX
+} CameraFuncType;
+
+typedef enum {
+    /* 0x00 */ CAM_DATA_00,
+    /* 0x01 */ CAM_DATA_01,
+    /* 0x02 */ CAM_DATA_02,
+    /* 0x03 */ CAM_DATA_03,
+    /* 0x04 */ CAM_DATA_04,
+    /* 0x05 */ CAM_DATA_05,
+    /* 0x06 */ CAM_DATA_06,
+    /* 0x07 */ CAM_DATA_07,
+    /* 0x08 */ CAM_DATA_08,
+    /* 0x09 */ CAM_DATA_09,
+    /* 0x0A */ CAM_DATA_10,
+    /* 0x0B */ CAM_DATA_11,
+    /* 0x0C */ CAM_DATA_12,
+    /* 0x0D */ CAM_DATA_13,
+    /* 0x0E */ CAM_DATA_14,
+    /* 0x0F */ CAM_DATA_15,
+    /* 0x10 */ CAM_DATA_16,
+    /* 0x11 */ CAM_DATA_17,
+    /* 0x12 */ CAM_DATA_18,
+    /* 0x13 */ CAM_DATA_19,
+    /* 0x14 */ CAM_DATA_20,
+    /* 0x15 */ CAM_DATA_21,
+} CameraDataType;
+
+typedef struct {
+    /* 0x0 */ s16 val;
+    /* 0x2 */ s16 param;
+} CameraModeValue; // size = 0x4
+
+typedef struct {
+    /* 0x0 */ s16 funcId;
+    /* 0x2 */ s16 numValues;
+    /* 0x4 */ CameraModeValue* values;
+} CameraMode; // size = 0x8
+
+/**
+ * Flags:
+ * (flags & 0xF): Priority (lower value has higher priority)
+ * (flags & 0x40000000): Store previous setting and bgCamData, also ignores water checks
+ * (flags & 0x80000000): Set camera setting based on bg/scene data and reset action function state
+ */
+typedef struct {
+    /* 0x0 */ u32 validModes;
+    /* 0x4 */ u32 flags;
+    /* 0x8 */ CameraMode* cameraModes;
+} CameraSetting; // size = 0xC
+
+typedef struct Camera {
+    /* 0x000 */ u8 paramData[0x50]; // function Data, acts like a heap that's reset every time a new action function is switched to
+    /* 0x050 */ Vec3f at;
+    /* 0x05C */ Vec3f eye;
+    /* 0x068 */ Vec3f up;
+    /* 0x074 */ Vec3f eyeNext;
+    /* 0x080 */ Vec3f quakeOffset;
+    /* 0x08C */ struct PlayState* play;
+    /* 0x090 */ Actor* trackActor; // the actor the camera is focused on. Most often Player, but can be any actor
+    /* 0x094 */ PosRot trackActorPosRot;
+    /* 0x0A8 */ Actor* target; // the actor that is being z-targeted
+    /* 0x0AC */ PosRot targetPosRot;
+    /* 0x0C0 */ f32 rUpdateRateInv;
+    /* 0x0C4 */ f32 pitchUpdateRateInv;
+    /* 0x0C8 */ f32 yawUpdateRateInv;
+    /* 0x0CC */ f32 yOffsetUpdateRate;
+    /* 0x0D0 */ f32 xzOffsetUpdateRate;
+    /* 0x0D4 */ f32 fovUpdateRate;
+    /* 0x0D8 */ f32 xzSpeed;
+    /* 0x0DC */ f32 dist;
+    /* 0x0E0 */ f32 speedRatio;
+    /* 0x0E4 */ Vec3f atActorOffset;
+    /* 0x0F0 */ Vec3f trackActorOffset;
+    /* 0x0FC */ f32 fov;
+    /* 0x100 */ f32 atLERPStepScale;
+    /* 0x104 */ f32 playerFloorHeight;
+    /* 0x108 */ Vec3f floorNorm;
+    /* 0x114 */ f32 waterYPos;
+    /* 0x118 */ s32 waterPrevBgCamDataId;
+    /* 0x11C */ s32 waterPrevCamSetting;
+    /* 0x120 */ s16 waterQuakeId;
+    /* 0x122 */ s16 unk122;
+    /* 0x124 */ void* data0;
+    /* 0x128 */ void* data1;
+    /* 0x12C */ s16 data2;
+    /* 0x12E */ s16 data3;
+    /* 0x130 */ s16 uid;
+    /* 0x132 */ UNK_TYPE1 pad132;
```

Comment:
This looks like the wrong type: an array of 2? Does it have a corresponding OoT var?

Reply:
It's just left as:
```
    /* 0x130 */ s16 uid;
    /* 0x132 */ char unk_132[2];
```
in OoT, so I guess I can sync it to that:
`/* 0x132 */ UNK_TYPE1 unk_132[2];`